File: src/main/java/corp/phonebook/data/repository/AuthRepository.java
package corp.phonebook.data.repository;

import org.springframework.stereotype.Repository;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.Optional;
import corp.phonebook.data.entity.User;

@Repository
public interface AuthRepository extends JpaRepository<User, Long> {
    Optional<User> findByEmail(String email);

    Optional<User> findByToken(String token);

    Optional<User> findBySessionId(String sessionId);
}

File: src/main/kotlin/corp/phonebook/data/repository/AuthRepository.kt
package corp.phonebook.data.repository

import org.springframework.stereotype.Repository
import org.springframework.data.jpa.repository.JpaRepository
import corp.phonebook.data.entity.User
import java.util.Optional

@Repository
interface AuthRepository : JpaRepository<User, Long> {
    fun findBySessionId(sessionId: String): Optional<User>
    fun findByEmail(email: String): Optional<User>
}